extensions/notebook/src/jupyter/serverInstance.ts

```diff
@@ -204,12 +204,19 @@ export class PerFolderServerInstance implements IServerInstance {
 	 * started when the log message with URL to connect to is emitted.
 	 */
 	protected async startInternal(): Promise<void> {
+		let startCommand: string;
+		let notebookConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(constants.notebookConfigKey);
+		let allowRoot: boolean = notebookConfig.get(constants.allowRoot);
 		if (this.isStarted) {
 			return;
 		}
 		let notebookDirectory = this.getNotebookDirectory();
 		this._token = await utils.getRandomToken();
-		let startCommand = `"${this.options.install.pythonExecutable}" "${this.notebookScriptPath}" --no-browser --ip=127.0.0.1 --allow-root --no-mathjax --notebook-dir "${notebookDirectory}" --NotebookApp.token=${this._token}`;
+		if (allowRoot) {
+			startCommand = `"${this.options.install.pythonExecutable}" "${this.notebookScriptPath}" --no-browser --ip=127.0.0.1 --allow-root --no-mathjax --notebook-dir "${notebookDirectory}" --NotebookApp.token=${this._token}`;
```

Comment:
Can make this a little more simple by doing something like

```
const allowRootParam = allowRoot ? '--allow-root' : '';
```

And then reference `${allowRootParam}` in the `startCommand`